sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3082,7 +3082,10 @@ object functions {
    * @since 1.5.0
    */
   def array_contains(column: Column, value: Any): Column = withExpr {
-    ArrayContains(column.expr, Literal(value))
+    value match {
```

Comment:
On second thoughts, we should use `lit()`, e.g., `ArrayContains(column.expr, lit(value).expr)`? WDYT? @viirya @maropu @HyukjinKwon

Reply:
Yup, that's what I was thinking too from my glance.

Reply:
Yes, I think so.

Reply:
+1